moirae/extractors/ecm.py

```diff
@@ -256,3 +257,277 @@ def extract(self, dataset: CellDataset) -> Resistance:
         knots = self.interpolate_r0(dataset.tables['raw_data'])
         return Resistance(base_values=knots, soc_pinpoints=self.soc_points,
                           interpolation_style=self.interpolation_style)
+
+
+class RCExtractor(BaseExtractor):
+    """Estimate the values of a parallel resistor-capacitor couples (RC)
+    for an ECM of a battery as a function of state of charge (SOC)
+
+    Required data: RC extraction requires rest periods of at least
+    :attr:`min_rest` seconds across a reasonable SOC range.
+
+    Algorithm:
+        1. Locate rest periods of at least :attr:`min_rest` seconds
+        2. Assign an SOC to each measurement based on :attr:`capacity`
+        3. Fit :attr:`n_rc` exponential models to the IV(t) data
+        4. The scale and power parameters are converted to the RC parameters
+        5. Fit a 1-D smoothing cubic spline for voltage as a function of SOC,
+           placing knots at :attr:`soc_points`.
+        6. Evaluate the spline at SOC points requested by the user,
+           return as a :class:`~moirae.models.ecm.components.Resistance` object
+           using the :attr:`interpolation_style` type of spline.
+
+    Args:
+        soc_points: SOC points at which to extract R0 or (``int``) number of

```

Comment:
Could you document the other arguments to `__init__` here as well?

Reply:
Sure! I'm a bit confused with the difference between documenting the args in the docstring versus just below it. Something to do with pydantic?

Reply:
Short answer: Yes, pydantic.

Longer answer: Both are fine and equivalent to the tool which renders our document. I prefer putting the documentation for the arguments in the docstring of the class (rather than for the `__init__` function) because pydantic auto-generates the `__init__` function. So, I try to use the class docstring so I can be consistent across all classes.